internal/app/starter/master_linux.go

```diff
@@ -20,93 +20,119 @@ import (
 	"github.com/sylabs/singularity/internal/pkg/util/mainthread"
 )
 
-// Master initializes a runtime engine and runs it
-func Master(rpcSocket, masterSocket int, isInstance bool, containerPid int, engine *engines.Engine) {
-	var fatal error
-	var status syscall.WaitStatus
+func doRPCthread(rpcSocket int, containerPid int, engine *engines.Engine, fatalChan chan error) {
```

Comment:
I would prefer _not_ to call these functions "thread".

Reply:
What do you want to call these functions then? I personally have no preference. Would just remove the term "thread" be okay?

Reply:
I don't have a good name for them. I thought about suggesting something like "masterService" or "rpcService"; or "initRPCserver". But I'm not sure if those are good names either, and I didn't want to nudge anyone in a particular direction.

I know they are not threads, though.

Reply:
@gvallee maybe just describe what they do ?

```
go createContainer(rpcSocket, containerPid, engine, fatalChan)
go startContainer(masterSocket, isInstance, containerPid, ppid, engine, fatalChan)
```

Reply:
that's good from my point of view